src/azure-cli-testsdk/azure/cli/testsdk/preparers.py

```diff
@@ -170,6 +170,18 @@ def remove_resource(self, name, **kwargs):
         if not self.dev_setting_sp_name:
             execute(self.cli_ctx, 'az ad sp delete --id {}'.format(self.result['appId']))
 
+# Function wise, enabling large payload recording has nothing to do with resource preparers 
+# We still base on it so that this decorator can chain with other preparers w/o too much hacks
+class LargeResponsePreparer(AbstractPreparer):
```

Comment:
Preparer is really meant for resources. I agree with your logic for using the AbstractPreparer, but I'd recommend a name like "AllowLargeResponse".